src/js/accordion/AccordionItem.js

```diff
@@ -0,0 +1,548 @@
+/**
+ * @file
+ * The Accordion Item class.
+ */
+
+import { isValidClassList, isValidType } from "../validate.js";
+import { addClass, removeClass } from "../domHelpers.js";
+
+/**
+ * Creates a new accordion item.
+ * @class
+ */
+class AccordionItem {
+
+  /**
+   * The HTML elements for the accordion item in the DOM.
+   *
+   * @protected
+   *
+   * @type {Object<HTMLElement>}
+   */
+  _dom = {
+    accordionItem: null,
+    controller: null,
+  };
+
+  /**
+   * A flag to determine if the accordion item is hidden.
+   *
+   * @protected
+   *
+   * @type {boolean}
+   */
+  _hidden = false;
+
+  /**
+   * The class to use to show the accordion item.
+   *
+   * @protected
+   *
+   * @type {string|string[]}
+   */
+  _showClass = "";
+
+  /**
+   * The class to use to hide the accordion item.
+   *
+   * @protected
+   *
+   * @type {string|string[]}
+   */
+  _hideClass = "";
+
+  /**
+   * The class to use when transitioning the accordion item.
+   *
+   * @protected
+   *
+   * @type {string|string[]}
+   */
+  _transitionClass = "";
+
+  /**
+   * The time in milliseconds the transition will take.
+   *
+   * @protected
+   *
+   * @type {number}
+   */
+  _transitionTimer = 150;
+
+  /**
+   * A flag to allow focus on up or down arrows.
+   *
+   * @protected
+   *
+   * @type {boolean}
+   */
+  _focusOnArrow = false;
```

Comment:
Call this _optionalSupport https://accessible-menu.dev/api/disclosure-menu.html#property--optionalsupport

And then the option for users would be "optionalKeySupport"

Reply:
Made this change